compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -145,8 +145,16 @@ impl<'local, 'context> Interpreter<'local, 'context> {
     }
 }
 
-fn failing_constraint<T>(message: impl Into<String>, location: Location) -> IResult<T> {
-    Err(InterpreterError::FailingConstraint { message: Some(message.into()), location })
+fn failing_constraint<T>(
+    message: impl Into<String>,
+    location: Location,
+    call_stack: &[Location],
+) -> IResult<T> {
+    Err(InterpreterError::FailingConstraint {
+        message: Some(message.into()),
+        location,
+        call_stack: call_stack.to_vec(),
```

Comment:
Can you store a `im::Vector` instead of a `Vec` for the failing constraint here? That way we can clone much more cheaply.

Reply:
Done! I didn't know if you wanted `im::Vector` for the call stack that's now on the Elaborator.